src/main/java/io/split/fastly/client/FastlyApiClient.java

```diff
@@ -110,6 +121,18 @@ public FastlyApiClient(final String apiKey, final String serviceId, AsyncHttpCli
         return purgeKey(key, buildHeaderForSoftPurge(extraHeaders));
     }
 
+    public Future<Response> softPurgeKeys(List<String> keys) {
```

Comment:
I suggest adding a validation for `keys.size() > 256` here and throwing an error if that's the input, based on this limitation from the docs: `Up to 256 surrogate keys can be purged in one batch request.`

Reply:
Following up what TC mentioned: 
```
Preconditions.checkNotNull(keys, "Keys cannot be null!");
Preconditions.checkArgument(keys.size() <= 256, "Cannot invalidate more than 256 keys!");
```

Reply:
added

Reply:
done @gpochettino